autogen/agentchat/agent.py

```diff
@@ -56,15 +67,32 @@ def generate_reply(
 
     async def a_generate_reply(
         self,
-        messages: Optional[List[Dict]] = None,
+        messages: Optional[List[Dict[str, Any]]] = None,
         sender: Optional["Agent"] = None,
-        **kwargs,
-    ) -> Union[str, Dict, None]:
-        """(Abstract async method) Generate a reply based on the received messages.
+        exclude: Optional[List[Callable[..., Any]]] = None,
+        **kwargs: Any,
+    ) -> Union[str, Dict[str, Any], None]:
+        """(Async) Generate a reply based on the received messages.
 
         Args:
             messages (list[dict]): a list of messages received.
             sender: sender of an Agent instance.
         Returns:
             str or dict or None: the generated reply. If None, no reply is generated.
         """
+
+
+@runtime_checkable
+class LLMAgent(Agent, Protocol):
+    """(In preview) A protocol for an LLM agent."""
+
+    @property
+    def system_message(self) -> str:
```

Comment:
Note that the system_message is not necessary a str, which could be a list of dict for multimodal with GPT-4. Similarly, it is also true for update_system_message. 

However, it is a discussion that should we consider multimodal here or consider multimodal in the subclass of ConversableAgent.

Reply:
Yes. Because the currently implementation of `ConversableAgent` assumes system message to be a string, so this PR simply makes it clear. We can make additional support for dict of list in future PRs.